framework/src/problems/FEProblemBase.C

```diff
@@ -4245,6 +4245,77 @@ FEProblemBase::execMultiAppTransfers(ExecFlagType type, Transfer::DIRECTION dire
              << string_direction << "MultiApps" << COLOR_DEFAULT << std::endl;
+
+    // output transfer information
+    unsigned int max_name_length = 4;
+    unsigned int max_type_length = 4;
+    unsigned int max_multiapp_name_length = 13;
+    for (const auto & transfer : transfers)
+    {
+      if (transfer->name().length() > max_name_length)
+        max_name_length = transfer->name().length();
+      if (transfer->type().length() > max_type_length)
+        max_type_length = transfer->type().length();
+      auto multiapp_transfer = dynamic_cast<MultiAppTransfer *>(transfer.get());
+      if (multiapp_transfer)
+      {
+        if (direction == MultiAppTransfer::TO_MULTIAPP)
+        {
+          unsigned int l = multiapp_transfer->getToMultiApp()->name().length();
+          if (l > max_multiapp_name_length)
+            max_multiapp_name_length = l;
+        }
+        else if (direction == MultiAppTransfer::FROM_MULTIAPP)
+        {
+          unsigned int l = multiapp_transfer->getFromMultiApp()->name().length();
+          if (l > max_multiapp_name_length)
+            max_multiapp_name_length = l;
+        }
+        else
+        {
+          unsigned int l1 = multiapp_transfer->getFromMultiApp()->name().length();
+          unsigned int l2 = multiapp_transfer->getToMultiApp()->name().length();
+          if (l1 + l2 + 2 > max_multiapp_name_length)
+            max_multiapp_name_length = l1 + l2 + 2;
+        }
+      }
+    }
+    unsigned int space = 2;
+    _console << std::left << std::setw(max_name_length + space) << "Name";
+    _console << std::left << std::setw(max_type_length + space) << "Type";
+    _console << std::left << std::setw(max_multiapp_name_length) << "MultiApp name\n";
+    unsigned int total_length =
+        max_name_length + max_type_length + max_multiapp_name_length + space * 2;
+    for (unsigned int i = 0; i < total_length; ++i)
+      _console << '-';
+    _console << '\n';
+
+    for (const auto & transfer : transfers)
+    {
+      _console << std::left << std::setw(max_name_length + space) << transfer->name();
+      _console << std::left << std::setw(max_type_length + space) << transfer->type();
+      auto multiapp_transfer = dynamic_cast<MultiAppTransfer *>(transfer.get());
+      if (multiapp_transfer)
+      {
+        std::string name;
+        if (direction == MultiAppTransfer::TO_MULTIAPP)
+          name = multiapp_transfer->getToMultiApp()->name();
+        else if (direction == MultiAppTransfer::FROM_MULTIAPP)
+          name = multiapp_transfer->getFromMultiApp()->name();
+        else
+        {
+          name += multiapp_transfer->getFromMultiApp()->name();
+          name += "=>";
+          name += multiapp_transfer->getToMultiApp()->name();
+        }
+        _console << std::left << std::setw(max_multiapp_name_length) << name;
+      }
+      _console << "\n";
+    }
+    for (unsigned int i = 0; i < total_length; ++i)
+      _console << '-';
+    _console << std::endl;
```

Comment:
this whole thing should be carefully reviewed but otherwise I think PR's good

Reply:
Let me do a quick run to paste the screen print-out before and after here.

Reply:
From something like this:
```
Starting Transfers on TIMESTEP_BEGIN To MultiApps
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid0
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid0
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid1
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid1
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid2
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid2
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid3
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid3
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid4
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid4
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid5
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid5
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid6
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid6
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid7
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid7
Beginning VariableValueSamplePostprocessorTransfer pebble_send_Tsolid8
Finished VariableValueSamplePostprocessorTransfer pebble_send_Tsolid8
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd0
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd0
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd1
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd1
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd2
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd2
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd3
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd3
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd4
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd4
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd5
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd5
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd6
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd6
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd7
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd7
Beginning VariableValueSamplePostprocessorTransfer pebble_send_ppd8
Finished VariableValueSamplePostprocessorTransfer pebble_send_ppd8
Transfers on TIMESTEP_BEGIN Are Finished

...

Starting Transfers on TIMESTEP_END From MultiApps
Beginning NearestNodeTransfer Tsolid_from_flow
Finished NearestNodeTransfer Tsolid_from_flow
Beginning NearestNodeTransfer Tfluid_from_flow
Finished NearestNodeTransfer Tfluid_from_flow
Beginning NearestNodeTransfer Rho_from_flow
Finished NearestNodeTransfer Rho_from_flow
Transfers on TIMESTEP_END Are Finished
```
to something
```
Starting Transfers on TIMESTEP_BEGIN From MultiApps
Name                     Type                                         MultiApp name
-----------------------------------------------------------------------------------
pebble_receive_T_mod_0   MultiAppPostprocessorNearestElementTransfer  pebble0      
pebble_receive_T_fuel_0  MultiAppPostprocessorNearestElementTransfer  pebble0      
pebble_receive_T_mod_1   MultiAppPostprocessorNearestElementTransfer  pebble1      
pebble_receive_T_fuel_1  MultiAppPostprocessorNearestElementTransfer  pebble1      
pebble_receive_T_mod_2   MultiAppPostprocessorNearestElementTransfer  pebble2      
pebble_receive_T_fuel_2  MultiAppPostprocessorNearestElementTransfer  pebble2      
pebble_receive_T_mod_3   MultiAppPostprocessorNearestElementTransfer  pebble3      
pebble_receive_T_fuel_3  MultiAppPostprocessorNearestElementTransfer  pebble3      
pebble_receive_T_mod_4   MultiAppPostprocessorNearestElementTransfer  pebble4      
pebble_receive_T_fuel_4  MultiAppPostprocessorNearestElementTransfer  pebble4      
pebble_receive_T_mod_5   MultiAppPostprocessorNearestElementTransfer  pebble5      
pebble_receive_T_fuel_5  MultiAppPostprocessorNearestElementTransfer  pebble5      
pebble_receive_T_mod_6   MultiAppPostprocessorNearestElementTransfer  pebble6      
pebble_receive_T_fuel_6  MultiAppPostprocessorNearestElementTransfer  pebble6      
pebble_receive_T_mod_7   MultiAppPostprocessorNearestElementTransfer  pebble7      
pebble_receive_T_fuel_7  MultiAppPostprocessorNearestElementTransfer  pebble7      
pebble_receive_T_mod_8   MultiAppPostprocessorNearestElementTransfer  pebble8      
pebble_receive_T_fuel_8  MultiAppPostprocessorNearestElementTransfer  pebble8      
-----------------------------------------------------------------------------------
Transfers on TIMESTEP_BEGIN Are Finished

...

Starting Transfers on TIMESTEP_END To MultiApps
Name                            Type                               MultiApp name
--------------------------------------------------------------------------------
to_nearest_streamline_transfer  MultiAppNearestStreamlineTransfer  streamline   
power_density_to_flow           MultiAppNearestNodeTransfer        flow         
Picard_Its                      MultiAppPostprocessorTransfer      flow         
--------------------------------------------------------------------------------
Transfers on TIMESTEP_END Are Finished
```
Note: looks like some transfers do not have the begin-finish prints, so I did not compare one to one.